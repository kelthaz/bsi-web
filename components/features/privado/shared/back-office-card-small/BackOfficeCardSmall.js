import PropTypes from 'prop-types';
import SvgVector from '../../../../svgs/SvgVector';
import SvgListaPequeña from '../../../../svgs/SvgListaPequeña';
import SvgReloj from '../../../../svgs/SvgReloj';
import styles from './back-office-card-small.module.scss';

const BackOfficeCardSmall = ({ iconoLista, title, subTitle, yellow }) => (
  <div>
    {yellow === false ? (
      <div className={`row ${iconoLista === true ? 'card-simple-blue-sky-small' : 'card-simple-blue-morning-small'}`}>
        <div className="col-4 mt-3">{iconoLista === true ? <SvgListaPequeña /> : <SvgVector />}</div>
        <div className="col-7 pr-0 pl-1 mt-2">
          <h4 className="body1 mb-0">{title}</h4>
          <div className="body3 overline">{subTitle}</div>
        </div>
      </div>
    ) : (
      <div className="row card-simple-yellow-summer-small">
        <div className="col-4 mt-3">
          <SvgReloj />
        </div>
        <div className="col-7 pr-0 pl-1 mt-2">
          <h4 className="body1 mb-0">{title}</h4>
          <div className="body3 overline">{subTitle}</div>
        </div>
      </div>
    )}
  </div>
);
// card-simple-yellow-summer-small
BackOfficeCardSmall.propTypes = {
  iconoLista: PropTypes.bool,
  yellow: PropTypes.bool,
  title: PropTypes.string,
  subTitle: PropTypes.string,
};

BackOfficeCardSmall.defaultProps = {
  iconoLista: true,
  yellow: false,
  title: null,
  subTitle: null,
};

export default BackOfficeCardSmall;
