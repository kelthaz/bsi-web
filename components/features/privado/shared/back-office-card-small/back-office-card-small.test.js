import { shallow } from 'enzyme';
import React from 'react';
import BackOfficeCardSmall from './BackOfficeCardSmall';

describe('Pruebas en el componente cards del backOffice small', () => {
  test('Debe de mostrarse correctamente el componente', () => {
    // arrange
    const props = { iconoLista: true, title: 'Prueba del titulo', subTitle: 'Prueba del subtitle' };
    // act
    const wrapper = shallow(<BackOfficeCardSmall {...props} />);
    // assert
    expect(wrapper).toMatchSnapshot();
  });
});
