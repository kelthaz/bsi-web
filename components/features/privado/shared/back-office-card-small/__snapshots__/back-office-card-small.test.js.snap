// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente cards del backOffice small Debe de mostrarse correctamente el componente 1`] = `
<div>
  <div
    className="row card-simple-blue-sky-small"
  >
    <div
      className="col-4 mt-3"
    >
      <SvgListaPequeÃ±a />
    </div>
    <div
      className="col-7 pr-0 pl-1 mt-2"
    >
      <h4
        className="body1 mb-0"
      >
        Prueba del titulo
      </h4>
      <div
        className="body3 overline"
      >
        Prueba del subtitle
      </div>
    </div>
  </div>
</div>
`;
